app/controllers/organisations/detail.js

```diff
@@ -2,5 +2,15 @@ import Ember from "ember";
 
 export default Ember.Controller.extend({
   gcOrganisationUsers: null,
-  backLinkPath: ""
+  backLinkPath: "",
+  organisationService: Ember.inject.service(),
+
+  actions: {
+    async searchOrganisation() {
+      const organisation = await this.get(
+        "organisationService"
+      ).userPickOrganisation();
+      this.replaceRoute("organisations.detail", organisation.get("id"));
```

Comment:
I'm seeing this in a few different files, maybe you could create a mixin with some navigation presets ? 

```suggestion
    async navigateToOrganisation(org = null) {
      const organisation = org || await this.get(
        "organisationService"
      ).userPickOrganisation();
      this.replaceRoute("organisations.detail", organisation.get("id"));
```

Also, using an optional argument like this ☝️  would make it so that whoever uses your mixin has the option of pre-selecting the organisation. 
In the context of a navigation mixin, I think a name like `navigateToOrganisation` is less likely to conflict with custom actions from controllers